tests/testthat/test-calculate_esi.R

```diff
@@ -0,0 +1,44 @@
+describe("calculate_esi", {
+  it("calculates ESI correctly for valid inputs", {
+    # Earth-like planet
+    expect_equal(
+      calculate_esi(1, 1),
+      1.0,
+      tolerance = 1e-6
+    )
+
+    expect_equal(
+      calculate_esi(0.6, 0.5),
+      0.7993933,
+      tolerance = 1e-6
+    )
+
+    expect_equal(
+      calculate_esi(14, 10^(-0.1) / 3.5^2),
+      0.2695072,
+      tolerance = 1e-6
+    )
```

Comment:
issue: Lets follow the AAA principles.

```suggestion
    expected1 = 1.0
    expected2 = 0.7993933
    expected3 = 0.2695072
    
    res1 = calculate_esi(1, 1)
    res2 = calculate_esi(0.6, 0.5)
    res3 = calculate_esi(14, 10^(-0.1) / 3.5^2)
    
    expect_equal(res1, expected1, tolerance = 1e-6)
    expect_equal(res2, expected2, tolerance = 1e-6)
    expect_equal(res3, expected3, tolerance = 1e-6)
```

we can sacrifice readability for performance in actual package / app, no reason to do that in tests.